File: Base/Workspace/Domain/Base/Client/Result.cs
namespace Allors.Workspace.Client
{
    using System.Collections.Generic;
    using System.Linq;

    using Data;

    public class Result
    {
        public Dictionary<string, SessionObject> Objects { get; }

        public Dictionary<string, SessionObject[]> Collections { get; }

        public Dictionary<string, object> Values { get; }

        public Result(Session session, PullResponse response)
        {
            this.Objects = response.namedObjects.ToDictionary(
                pair => pair.Key,
                pair => session.Get(long.Parse(pair.Value)));
            this.Collections = response.namedCollections.ToDictionary(
                pair => pair.Key,
                pair => pair.Value.Select(v => session.Get(long.Parse(v))).ToArray());
            this.Values = response.namedValues.ToDictionary(
                pair => pair.Key, 
                pair => pair.Value);
        }
    }
}

File: Core/Workspace/Typescript.legacy/Angular/Result.ts
/// <reference path="allors.module.ts" />
namespace Allors {
    export class Result {

        objects: { [name: string]: ISessionObject; } = {};
        collections: { [name: string]: ISessionObject[]; } = {};
        values: { [name: string]: any; } = {};

        constructor(session: ISession, response: Data.PullResponse) {
            _.map(response.namedObjects, (v, k) => {
                this.objects[k] = session.get(v);
            });

            _.map(response.namedCollections, (v, k) => {
                this.collections[k] = _.map(v, (obj) => { return session.get(obj) });
            });

            _.map(response.namedValues, (v, k) => {
                this.values[k] = v;
            });
        }
    }
}